src/strategies_and_params/poststepstrategy.jl

```diff
@@ -271,16 +271,12 @@ function single_particle_density(dvec; component=0)
     K = keytype(dvec)
     V = float(valtype(dvec))
     M = num_modes(K)
-    N = num_particles(K)
 
-    result = mapreduce(
-        +, pairs(dvec);
-        init=MultiScalar(ntuple(_ -> zero(V), Val(M)))
-    ) do (k, v)
+    result = sum(pairs(dvec); init=MultiScalar(ntuple(_ -> zero(V), Val(M)))) do (k, v)
         MultiScalar(v^2 .* single_particle_density(k; component))
```

Comment:
```suggestion
        MultiScalar(abs2(v) .* single_particle_density(k; component))
```
values could be complex